pennylane/templates/layers/gate_fabric.py

```diff
@@ -0,0 +1,221 @@
+    a 1-parameter 4-qubit spin-adapted spatial orbital rotation gate, which is implemented by :class:`~.OrbitalRotation()`
+    operation and 1-parameter 4-qubit diagonal pair-exchange gate, which is equivalent to the :class:`~.DoubleExcitation()`
+    operation. In addition to these two gates, the gate element :math:`\hat{Q}(\theta, \phi)` can also include an optional
+    constant :math:`\hat{\Pi} \in \{\hat{I}, \text{OrbitalRotation}(\pi)\}` gate.
+
+    .. figure:: ../../_static/templates/layers/q_gate_decompositon.png
+        :align: center
+        :width: 75%
+        :target: javascript:void(0);
+
+    |
+
+    The individual four qubit :class:`~.DoubleExcitation()` and :class:`~.OrbitalRotation()` gates given here are equivalent to the
+    :math:`\text{QNP}_{PX}(\theta)` and :math:`\text{QNP}_{OR}(\phi)` gates presented in `arXiv:2104.05692 <https://arxiv.org/abs/2104.05695>`_
+    respectively. Moreover, regardless of the choice of :math:`\hat{\Pi}`, this gate fabric will exactly preserve the quantum numbers
+    :math:`\hat{N}_{\alpha}`, :math:`\hat{N}_{\beta}` and :math:`\hat{S}^{2}`.
+
+    Args:
+        weights (tensor_like): Array of weights of shape ``(D, L, 2)``\,
+            where ``D`` is the number of gate fabric layers and ``L = N/2-1``
+            is the number of :math:`\hat{Q}(\theta, \phi)` gates per layer.
+        wires (Iterable): wires that the template acts on
+        init_state (tensor_like): iterable of shape ``(len(wires),)``\, representing the Hartree-Fock state in Jordan-Wigner basis
+            that is used to initialize the wires.
+        include_pi (boolean): If ``include_pi = True``\, the optional constant :math:`\hat{\Pi}` gate  is set to :math:`\text{OrbitalRotation}(\pi)`.
+            Default value is :math:`\hat{I}`.
+
+    .. UsageDetails::
+
+        #. The number of wires :math:`N` has to be equal to the number of
+           spin orbitals included in the active space, and should be even.
+
+        #. The number of trainable parameters scales linearly with the number of layers as
+           :math:`2 D (N/2-1)`.
+
+        An example of how to use this template is shown below:
+
+        .. code-block:: python
+
+            import numpy as np
+            import pennylane as qml
+            from pennylane.templates import GateFabric
+            from functools import partial
+
+            # Build the electronic Hamiltonian from a local .xyz file
+            h, qubits = qml.qchem.molecular_hamiltonian("h2", "h2.xyz")
+
+            # Define the Hartree-Fock state
+            electrons = 2
+            ref_state = qml.qchem.hf_state(electrons, qubits)
+
+            # Define the device
+            dev = qml.device('default.qubit', wires=qubits)
+
+            # Define the ansatz
+            ansatz = partial(GateFabric, init_state=ref_state, include_pi=True)
+
+            # Define the cost function
+            cost_fn = qml.ExpvalCost(ansatz, h, dev)
```

Comment:
Might consider having an example that works with `expval(H)`.

Reply:
Yes, I have changed the example now with one with `expval(H)`.